src/app/modules/main/profile_section/profile/models/profile_preferences_accounts_model.nim

```diff
@@ -0,0 +1,172 @@
+
+  method roleNames(self: ProfileShowcaseAccountsModel): Table[int, string] =
+    {
+      ModelRole.ShowcaseVisibility.int: "showcaseVisibility",
+      ModelRole.Order.int: "order",
+
+      ModelRole.Address.int: "address",
+      ModelRole.Name.int: "name",
+      ModelRole.WalletType.int: "walletType",
+      ModelRole.Emoji.int: "emoji",
+      ModelRole.ColorId.int: "colorId",
+    }.toTable
+
+  method data(self: ProfileShowcaseAccountsModel, index: QModelIndex, role: int): QVariant =
+    if (not index.isValid):
+      return
+
+    if (index.row < 0 or index.row >= self.items.len):
+      return
+
+    let item = self.items[index.row]
+    let enumRole = role.ModelRole
+
+    case enumRole:
+    of ModelRole.ShowcaseVisibility:
+      result = newQVariant(item.showcaseVisibility.int)
+    of ModelRole.Order:
+      result = newQVariant(item.order)
+    of ModelRole.Address:
+      result = newQVariant(item.address)
+    of ModelRole.Name:
+      result = newQVariant(item.name)
+    of ModelRole.WalletType:
+      result = newQVariant(item.walletType)
+    of ModelRole.Emoji:
+      result = newQVariant(item.emoji)
+    of ModelRole.ColorId:
+      result = newQVariant(item.colorId)
+
+  proc findIndexForAccount(self: ProfileShowcaseAccountsModel, address: string): int =
+    for index in 0 ..< self.items.len:
+      if (self.items[index].address == address):
+        return index
+    return -1
+
+  proc hasItemInShowcase*(self: ProfileShowcaseAccountsModel, address: string): bool {.slot.} =
+    let ind = self.findIndexForAccount(address)
+    if ind == -1:
+      return false
+    return self.items[ind].showcaseVisibility != ProfileShowcaseVisibility.ToNoOne
+
+  proc itemsUpdated*(self: ProfileShowcaseAccountsModel) {.signal.}
+
+  proc appendItem*(self: ProfileShowcaseAccountsModel, item: ProfileShowcaseAccountItem) =
+    let parentModelIndex = newQModelIndex()
+    defer: parentModelIndex.delete
+    self.beginInsertRows(parentModelIndex, self.items.len, self.items.len)
+    self.items.add(item)
+    self.endInsertRows()
+    self.countChanged()
+
+  proc insertOrUpdateItemImpl(self: ProfileShowcaseAccountsModel, item: ProfileShowcaseAccountItem) =
+    let ind = self.findIndexForAccount(item.address)
+    if ind == -1:
+      self.appendItem(item)
+    else:
+      self.items[ind] = item
+
+      let index = self.createIndex(ind, 0, nil)
+      defer: index.delete
+      self.dataChanged(index, index, @[
+        ModelRole.ShowcaseVisibility.int,
+        ModelRole.Order.int,
+        ModelRole.Address.int,
+        ModelRole.Name.int,
+        ModelRole.WalletType.int,
+        ModelRole.Emoji.int,
+        ModelRole.ColorId.int,
+      ])
+
+  proc insertOrUpdateItemJson(self: ProfileShowcaseAccountsModel, itemJson: string) {.slot.} =
+    self.insertOrUpdateItemImpl(itemJson.parseJson.toProfileShowcaseAccountItem())
+
+  proc insertOrUpdateItem*(self: ProfileShowcaseAccountsModel, item: ProfileShowcaseAccountItem) =
```

Comment:
I'm not sure why you need this function as well. I see that you call `itemsUpdated`, but is that necessary? The model will update already with `dataChanged` no?

Reply:
i've minimised signals in the last commit here